### PR TITLE
RUN-1230 - Add UiSocket support for execution log

### PR DESCRIPTION
**Pull request by ahormazabal**
This PR:
- Adds support for using the `UiSocket` component within the execution log UI, enabling UI Plugins to safely inject content into the log viewer.
- Adds the `metadata` field to execution log API output, in order to get additional log entry metadata.

### CODE REVIEW

File: rundeckapp/grails-spa/packages/ui-trellis/src/components/execution-log/logBuilder.ts [hunk 1]
```diff
@@ -134,36 +138,36 @@ export class LogBuilder {
...
-    }
+    const vue = new EntryFlex({
+      propsData: {
+        eventBus: this.eventBus,
+        selected: selected,
+        timestamps: this.opts.time.visible,
+        command: this.opts.command.visible,
+        gutter: this.opts.gutter.visible,
+        nodeBadge: this.opts.nodeIcon,
+        lineWrap: this.opts.content.lineWrap,
```
Comment (gschueler) [inline rundeckapp/grails-spa/packages/ui-trellis/src/components/execution-log/logBuilder.ts:153]: could reduce this to just passing the opts directly I suppose

File: rundeckapp/grails-spa/packages/ui-trellis/src/components/utils/UiSocket.vue [hunk 2]
```diff
@@ -15,8 +15,9 @@ import {Component, Prop, Watch} from 'vue-property-decorator'
 
 import {
     getRundeckContext,
-} from '@rundeck/ui-trellis'
-import {UIItem, UIWatcher} from '@rundeck/ui-trellis/lib/stores/UIStore'
+} from '../../rundeckService'
+import {UIItem, UIWatcher} from '../../stores/UIStore'
+// import {UIItem, UIWatcher} from '@rundeck/ui-trellis/lib/stores/UIStore'
```
Comment (gschueler) [inline rundeckapp/grails-spa/packages/ui-trellis/src/components/utils/UiSocket.vue:20]:
```suggestion
```

File: rundeckapp/grails-spa/packages/ui-trellis/src/stores/ExecutionOutput.ts [hunk 3]
```diff
@@ -118,14 +118,33 @@ export class ExecutionOutput {
...
+                maxlines: maxLines
+            }
+        }).then(response => {
+            if (response.status != 200) {
+                throw new Error("Error calling execution log api: " + response.bodyAsText)
+            }
+            return response.parsedBody
+        })
+    }
+
```
[the remark this answers is not in the document]
Comment (ahormazabal) [reply rundeckapp/grails-spa/packages/ui-trellis/src/stores/ExecutionOutput.ts]: I'm not sure if i'm getting it. This is not a vue component but indeed a utility.

[the remark this answers is not in the document]
Comment (gschueler) [reply rundeckapp/grails-spa/packages/ui-trellis/src/stores/ExecutionOutput.ts]: my mistake, I thought i was looking at a vue file

**Review by gschueler — APPROVED** [review]

**Pull request merged**